Tests/YamsTests/PerformanceTests.swift

```diff
@@ -57,90 +57,137 @@ class PerformanceTests: XCTestCase {
         "/SourceKitten/Source/SourceKittenFramework/Xcode.swift"
     ]
 
-    func testSourceKittenIssue289UsingLoad() {
-        guard let yamlString = try? String(contentsOfFile: filename, encoding: .utf8) else {
+    func loadYAML() throws -> String {
+        let data = try Data(contentsOf: URL(fileURLWithPath: filename))
+        return String(data: data, encoding: .utf8)!
+    }
+
+    func parseSourceKittenIssue289UsingLoad(yaml: String, spmName: String, encoding: Parser.Encoding) {
+        do {
+            guard let object = try Yams.load(yaml: yaml, .default, .default, encoding) as? [String: Any],
+                let commands = (object["commands"] as? [String: [String: Any]])?.values,
+                let moduleCommand = commands.first(where: { ($0["module-name"] as? String ?? "") == spmName }),
+                let imports = moduleCommand["import-paths"] as? [String],
+                let otherArguments = moduleCommand["other-args"] as? [String],
+                let sources = moduleCommand["sources"] as? [String] else {
+                    XCTFail("Invalid result form Yams.load()")
+                    return
+            }
+            XCTAssertEqual(imports, self.expectedImports)
+            XCTAssertEqual(otherArguments, self.expectedOtherArguments)
+            XCTAssertEqual(sources, self.expectedSources)
+        } catch {
+            XCTFail("\(error)")
+        }
+    }
+
+    func testUsingLoadWithUTF16() {
```

Comment:
XCTest tests can be marked as `throws`, and if a test throws, it's marked as a test failure.

```swift
func testUsingLoadWithUTF16() throws {
    let yaml = try loadYAML()
    self.measure {
        parseSourceKittenIssue289UsingLoad(yaml: yaml, spmName: "SourceKittenFramework", encoding: .utf16)
    }
}
```